update/genesis/export.go

```diff
@@ -372,6 +374,17 @@ func (se *stateExport) exportNodesSetupJson(validators map[uint32][]*state.Valid
 		}
 	}
 
+	sort.Slice(initialNodes, func(i, j int) bool {
```

Comment:
This will order in descending way (Shard n, Shard n-1, Shard 0) and also the PubKeys. It does matter?
Anyway you can simplify the code between lines 377-386 by replacing it with:

```
	sort.Slice(initialNodes, func(i, j int) bool {
		if initialNodes[i].AssignedShard() == initialNodes[j].AssignedShard() {
			return bytes.Compare(initialNodes[i].PubKeyBytes(), initialNodes[j].PubKeyBytes()) < 0
		}
		return initialNodes[i].AssignedShard() < initialNodes[j].AssignedShard()
	})

```

Reply:
refactored. now it sorts after public keys. there were some issues with the previous implementation